exotations/task_maps/task_map/src/JointLimit.cpp

```diff
@@ -79,6 +70,16 @@ void JointLimit::update(Eigen::VectorXdRefConst x, Eigen::VectorXdRef phi)
 {
     if (phi.rows() != N) throw_named("Wrong size of phi!");
     phi.setZero();
+
+    const Eigen::MatrixXd limits = scene_->getSolver().getJointLimits();
+
+    for (int i = 0; i < N; i++)
```

Comment:
Couple of changes required:
(a) `low_limits_` and `high_limits_` and `tau_` do not strictly have to be member variables (but perhaps good for no alloations)
(b) Replace with (vector operations)

```cpp
Eigen::VectorXdRefConst low_limits = limits.col(0);
Eigen::VectorXdRefConst high_limits = limits.col(1);
tau_ = 0.5 * safe_percentage_ * (high_limits - low_limits);
```